framework/src/modules/chain/blocks/blocks.js

```diff
@@ -449,6 +456,120 @@ class Blocks extends EventEmitter {
 		});
 	}
 
+	async _receiveBlockFromNetworkV2(block) {
+		// Current time since Lisk Epoch
+		// Better to do it here rather than in the Sequence so receiving time is more accurate
+		const newBlockReceivedAt = this.slots.getTime();
+
+		// Execute in sequence via sequence
+		return this.sequence.add(callback => {
+			try {
+				this._shouldNotBeActive();
+			} catch (error) {
+				callback(error);
+				return;
+			}
+			this._isActive = true;
+			this._forkChoiceTask(block, newBlockReceivedAt)
+				.then(result => callback(null, result))
+				.catch(error => callback(error))
+				.finally(() => {
+					this._isActive = false;
+				});
+		});
+	}
+
+	// PRE: Block has been validated and verified before
+	async _processReceivedBlock(block) {
+		this._updateLastReceipt();
+		try {
+			const newBlock = await this.blocksProcess.processBlock(
+				block,
+				this._lastBlock,
+				validBlock => this.broadcast(validBlock)
+			);
+
+			this.logger.debug(
+				`Successfully applied new received block id: ${block.id} height: ${
+					block.height
+				} round: ${this.slots.calcRound(
+					block.height
+				)} slot: ${this.slots.getSlotNumber(block.timestamp)} reward: ${
+					block.reward
+				} version: ${block.version}`
+			);
+
+			await this._updateBroadhash();
+			this._lastBlock = newBlock;
+			this._isActive = false;
+		} catch (error) {
+			this.logger.error(
+				`Failed to apply new received block id: ${block.id} height: ${
+					block.height
+				} round: ${this.slots.calcRound(
+					block.height
+				)} slot: ${this.slots.getSlotNumber(block.timestamp)} reward: ${
+					block.reward
+				} version: ${block.version} with error: ${error}`
+			);
+			this._isActive = false;
+			throw error;
+		}
```

Comment:
Add a block `finally` to set `this._isActive = false;` or similar postrequisits. That will make it more understandable and align with promise convention.